### ISSUE TITLE
[4주차] soft state 예시?!

### ISSUE THREAD

**Issue by jiwooo-kim** [kiworkshop/designing-data-intensive-applications#29]
p.224 Soft state 뭔지 더 이해하고 싶어 예시가 궁금하네용

**Comment by myeongjae-kim** [comment]
Certainly! The term "BASE" is often used in the context of NoSQL databases to contrast with the more traditional "ACID" properties of relational databases.

ACID stands for:

Atomicity: All operations in a transaction are completed or none are.
Consistency: The database starts from one consistent state and ends in another consistent state.
Isolation: Concurrent execution of transactions leaves the database in the same state as if the transactions were executed sequentially.
Durability: Once a transaction is committed, its results are permanent and can survive any subsequent failures.
On the other hand, BASE stands for:

Basically Available: The system does guarantee availability.
Soft state: The state of the system may change over time, even without input.
Eventually consistent: The system will become consistent over time, but at any given point, it may be inconsistent.
"Soft state" in the context of BASE means that the system's state is allowed to be temporary or intermediate. Given enough time without new inputs, it might change. This contrasts with traditional systems where the state remains fixed until there's an external change (like a transaction).

The idea behind "soft state" is to permit temporary inconsistencies in exchange for higher availability or resilience in distributed systems. For example, in a distributed database, if one node hasn't yet received an update, it might temporarily be in an inconsistent state with other nodes. However, as updates propagate and systems synchronize, they eventually reach a consistent state.

In essence, while ACID prioritizes strict consistency at all times (at the potential expense of availability during network partitions or failures), BASE prioritizes availability and accepts that there might be some periods of inconsistency, with the system eventually converging towards consistency.

**Comment by harrisleesh** [comment]
지우쌤 : Soft-state는 NoSQL의 BASE 성질 중 하나로, 시스템의 상태가 입력 없이도 시간이 지나면서 변할 수 있음을 의미해요. 즉, 네트워크의 상태가 일정하지 않고 변할 수 있다는 것을 의미하며, 이는 시스템이 항상 일관성을 유지하지 않고 일시적으로 비일관성 상태를 허용한다는 것을 의미해요.
